code/modules/mob/living/simple_animal/bot/firebot.dm

```diff
@@ -142,20 +133,20 @@
 		internal_ext.max_water = INFINITY
 		internal_ext.refill()
 
-/mob/living/simple_animal/bot/firebot/Topic(href, href_list)
+/mob/living/simple_animal/bot/firebot/ui_act(action, params)
 	if(..())
-		return TRUE
-
-	switch(href_list["operation"])
+		return
+	if(!issilicon(usr) && !IsAdminGhost(usr) && !(bot_core.allowed(usr) || !locked))
+		return
```

Comment:
This check is incorrect. If you are a ghost you can never be a silicon. False && True is always false. This check will never pass and thus, you will always be allowed to interact with this.

Reply:
replaced such checks with the following, as per suggestion
```dm
	if(!(bot_core.allowed(usr) || usr.has_unlimited_silicon_privilege) || locked)
		return
```
I think this should work, bar for requiring aghosts to unlock the interface before using it, but that should be possible with the ``bot.dm`` implementation of the proc which is called before